app/controllers/invitations_controller.rb

```diff
@@ -0,0 +1,24 @@
+class InvitationsController < ApplicationController
+  before_action :require_user
+
+  def new
+    @invitation = Invitation.new
+  end
+
+  def create
+    @invitation = Invitation.new(invitation_params.merge!(inviter: current_user))
+
+    if @invitation.save
+      AppMailer.invitation_email(@invitation).deliver
+      flash[:success] = "You have successfully invited #{@invitation.recipient_name}"
+      redirect_to new_invitation_path
+    else
+      flash[:danger] = 'Please enter all details'
```

Comment:
Whenever you use `render`, you need to use `flash`. `flash.now` survives the next request as well.